src/parser/jobs/drg/Procs.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+export default class Procs extends Module {
+	static handle = 'procs'
+	static dependencies = [
+		'downtime',
+		'suggestions',
+	]
+
+	_fangAndClaw = {
+		casts: 0,
+		removed: 0,
+	}
+	_wheelingThrust = {
+		casts: 0,
+		removed: 0,
+	}
+	_mirageDive = {
+		casts: 0,
+		removed: 0,
+		overwritten: 0,
+	}
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.FANG_AND_CLAW.id}, this._onFangAndClaw)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.WHEELING_THRUST.id}, this._onWheelingThrust)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.MIRAGE_DIVE.id}, this._onMirageDive)
+		this.addHook('refreshbuff', {by: 'player', abilityId: STATUSES.DIVE_READY.id}, this._onReadyOverwritten) // The other two can't be overwritten due to how they drop
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.SHARPER_FANG_AND_CLAW.id}, this._onSharperRemoved)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.ENHANCED_WHEELING_THRUST.id}, this._onEnhancedRemoved)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.DIVE_READY.id}, this._onReadyRemoved)
+		this.addHook('complete', this._onComplete)
+	}
+
+	// For all of our cast/removal tracking, we only want to know if it happened outside of downtime to avoid errant penalization.
+	// The only one that doesn't need the check is overwriting Mirage Dive, since you can't get a proc from an invulnerable target.
+	_onFangAndClaw(event) {
+		if (!this.downtime.isDowntime(event.timestamp)) {
+			this._fangAndClaw.casts++
+		}
+	}
+
+	_onWheelingThrust(event) {
+		if (!this.downtime.isDowntime(event.timestamp)) {
+			this._wheelingThrust.casts++
+		}
+	}
+
+	_onMirageDive(event) {
+		if (!this.downtime.isDowntime(event.timestamp)) {
+			this._mirageDive.casts++
+		}
+	}
+
+	_onReadyOverwritten() {
+		this._mirageDive.overwritten++
+	}
+
+	_onSharperRemoved(event) {
+		if (!this.downtime.isDowntime(event.timestamp)) {
+			this._fangAndClaw.removed++
+		}
+	}
+
+	_onEnhancedRemoved(event) {
+		if (!this.downtime.isDowntime(event.timestamp)) {
+			this._wheelingThrust.removed++
+		}
+	}
+
+	_onReadyRemoved(event) {
+		if (!this.downtime.isDowntime(event.timestamp)) {
+			this._mirageDive.removed++
+		}
+	}
+
+	_onComplete() {
+		const droppedFang = this._fangAndClaw.removed - this._fangAndClaw.casts
+		const droppedWheeling = this._wheelingThrust.removed - this._wheelingThrust.casts
+		const droppedMirage = this._mirageDive.removed - this._mirageDive.casts
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: ACTIONS.FANG_AND_CLAW.icon,
+			content: <Trans id="drg.procs.suggestions.fang.content">
```

Comment:
This is duping up a bit with the suggestion below it - is there a particularly strong case to keep them apart? Might be able to tier it akin to `smn/DoTs` - it uses the worse of the cases as the tier `value`, and mentions both values in the `why`.

Reply:
I could see a case for merging them. My only concern is that the `why` might be goofy if I include them both, since in most of the parses I've looked at, the numbers are 1 and 0 for dropped proc counts. It's less of an issue for SMN, since you very rarely have 0s of DoT clipping, but here it would wind up as "You dropped 1 Enhanced... proc and 0 Sharper... procs." If you think that sounds fine I'll go ahead and do it (although I'll probably use the sum for the tiers rather than the max).

Reply:
Mmm.

You might be able to use plural rules to hide one or the other, but I'd be fearful of how that'd hold up in other languages which _kinda_ defeats the purpose of i18n.

I'm not bothered by 0s as long as it's obvious why it _is_ there, but comes down to taste. Your call.